internal/core/src/common/QueryResult.h

```diff
@@ -228,6 +228,7 @@ struct RetrieveResult {
     void* segment_;
     std::vector<int64_t> result_offsets_;
     std::vector<DataArray> field_data_;
+    bool has_more_result;
 };
```

Comment:
actually, this should be 'maybe_has_more_result', when it's true, we cannot discard all possible following results from this source, when it's false, it's definitely ok to ignore results from this source.